stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+
+        _NODISCARD constexpr sys_seconds date() const noexcept {
+            return _Date;
+        }
+
+        _NODISCARD constexpr seconds value() const noexcept {
+            return _Is_positive ? seconds{1} : seconds{-1};
+        }
+
+        _NODISCARD constexpr bool _Positive() const noexcept {
+            return _Is_positive;
+        }
+
+    private:
+        sys_seconds _Date;
+        bool _Is_positive;
+    };
+
+    _NODISCARD inline bool operator==(const leap_second& _Left, const leap_second& _Right) noexcept {
+        return _Left.date() == _Right.date();
+    }
+    template <class _Duration>
+    _NODISCARD bool operator==(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Left.date() == _Right;
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator<(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Left.date() < _Right;
+    }
+    template <class _Duration>
+    _NODISCARD bool operator<(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return _Left < _Right.date();
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator>(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Right < _Left.date();
+    }
+    template <class _Duration>
+    _NODISCARD bool operator>(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return _Right.date() < _Left;
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator<=(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return !(_Right < _Left.date());
+    }
+    template <class _Duration>
+    _NODISCARD bool operator<=(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return !(_Right.date() < _Left);
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator>=(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return !(_Left.date() < _Right);
+    }
+    template <class _Duration>
+    _NODISCARD bool operator>=(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return !(_Left < _Right.date());
+    }
+
+#ifdef __cpp_lib_concepts
+    template <class _Duration>
+    requires three_way_comparable_with<sys_seconds, sys_time<_Duration>> constexpr auto operator<=>(
+        const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Left.date() <=> _Right;
+    }
+    constexpr inline strong_ordering operator<=>(const leap_second& _Left, const leap_second& _Right) noexcept {
+        return _Left.date() <=> _Right.date();
+    }
+#endif // __cpp_lib_concepts
+
+    inline pair<vector<leap_second>, bool> _Xtzdb_generate_leap_seconds(const size_t _Current_size) {
+        // Returns empty vector if no new leap seconds are found.
+        constexpr static leap_second _Pre_2018_leap_seconds[]{

```

Comment:
We conventionally say `static constexpr` for arrays like this. (The `static` is indeed valuable here.)